src/tokens/TokenImport.tsx

```diff
@@ -0,0 +1,193 @@
+  }, [tokenAddress])
+
+  const errorMessage = () => {
+    // TODO RET-892: when states and validation are added, choose appropriate error or return null
+    const errors = [
+      t('tokenImport.invalidToken'),
+      t('tokenImport.alreadySupported'),
+      t('tokenImport.alreadyImported'),
+    ]
+    return <Text style={styles.errorLabel}>{errors[0]}</Text>
+  }
+
+  return (
+    <SafeAreaView style={styles.container}>
+      <CustomHeader
+        style={styles.customHeader}
+        left={<BackButton />}
+        title={<Text style={styles.headerTitle}>{t('tokenImport.title')}</Text>}
+      />
+      <ScrollView>
+        <View style={styles.titleContainer}>
+          <InLineNotification
+            severity={Severity.Informational}
+            description={t('tokenImport.notification')}
+          />
+
+          <View style={{ gap: Spacing.Thick24, width: '100%' }}>
+            {/* Token Address */}
+            <View style={styles.textInputGroup}>
+              <Text style={styles.label}>{t('tokenImport.tokenAddress')}</Text>
+              <TextInput
+                onChangeText={setTokenAddress}
+                value={tokenAddress}
+                multiline={false}
+                style={styles.messageTextInput}
+                placeholderTextColor={Colors.gray4}
+                underlineColorAndroid="transparent"
+                numberOfLines={1}
+                placeholder={t('tokenImport.tokenAddressPlaceholder') ?? undefined}
+                showClearButton={true}
+                right={
+                  <PasteButton
+                    value={tokenAddress}
+                    setValue={setTokenAddress}
+                    onPress={() => ValoraAnalytics.track(AssetsEvents.tap_import_token_paste)}
+                  />
+                }
+                maxLength={42} // 0x prefix and 20 bytes
+              />
+            </View>
+
+            {/* Token Symbol */}
+            <View style={styles.textInputGroup}>
+              <Text style={styles.label}>{t('tokenImport.tokenSymbol')}</Text>
+              <TextInput
+                onChangeText={setTokenSymbol}
+                value={tokenSymbol}
+                multiline={false}
+                style={styles.messageTextInput}
+                placeholderTextColor={Colors.gray4}
+                underlineColorAndroid="transparent"
+                numberOfLines={1}
+                showClearButton={true}
+                editable={isValidAddress}
+                right={isValidAddress && <GreenLoadingSpinner height={32} />} // TODO RET-892: once loaded, hide the spinner
+              />
+              {errorMessage()}
+            </View>
+
+            {/* Network */}
+            <View style={styles.textInputGroup}>
+              <Text style={styles.label}>{t('tokenImport.network')}</Text>
+              <TextInput
+                onChangeText={setTokenSymbol}
+                value={t('celoGold') ?? undefined}
```

Comment:
you can use `NETWORK_NAMES` instead...doesn't seem right to use the name of the celo asset here